src/compiler/diagnosticMessages.json

```diff
@@ -8317,5 +8317,9 @@
     "Enum member following a non-literal numeric member must have an initializer when 'isolatedModules' is enabled.": {
         "category": "Error",
         "code": 18056
+    },
+    "Arbitrary module namespace identifiers are not supported when the '--module' flag is set to 'es2015' or 'es2020'.": {

```

Comment:
"Arbitrary module namespace identifiers" is a bit too much inside baseball for most developers. It might be worthwhile to report an error like this instead:

> String literal import and export names are not supported when the '--module' flag is ...